### ISSUE TITLE
Generated aboutlibraries.xml Contains { in Resource Name. Build Failure

### ISSUE THREAD

**Issue by StuStirling** [mikepenz/AboutLibraries#564]
## About this issue

When creating a release build of my app I'm getting the following error:

`.../app/build/generated/aboutlibraries/Release/res/values/aboutlibraries.xml: Resource and asset merger: '{' is not a valid resource name character`

Looking at the generated file I can see the following resources are defined.

```xml
<string name="define_plu_org_glassfish__${extension_name}" translatable="false"></string>
<string name="library_org_glassfish__${extension_name}_libraryName" translatable="false">${extension.name} API v.${spec.version}</string>
<string name="library_org_glassfish__${extension_name}_libraryDescription" translatable="false"><![CDATA[Common Annotations for the JavaTM Platform API version ${spec.version} Repackaged as OSGi bundle in GlassFish]]></string>
<string name="library_org_glassfish__${extension_name}_libraryVersion" translatable="false">10.0-b28</string>
<string name="library_org_glassfish__${extension_name}_libraryArtifactId" translatable="false">org.glassfish:${extension.name}:10.0-b28</string>
```

Glassfish is included in multiple different modules of my app but here is an example from the main module:

```groovy
compileOnly "org.glassfish:javax.annotation:$javaxAnnotationVersion"
```

## Details
Plugin version 8.6.0
Gradle build tools version 4.1.1

Android Studio 4.1.1
Build #AI-201.8743.12.41.6953283, built on November 5, 2020
Runtime version: 1.8.0_242-release-1644-b3-6915495 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.15.7
GC: ParNew, ConcurrentMarkSweep
Memory: 3987M
Cores: 16
Registry: ide.new.welcome.screen.force=true, external.system.auto.import.disabled=true
Non-Bundled Plugins: Key Promoter X, com.godwin.json.parser, org.jetbrains.kotlin, com.developerphil.adbidea

**Comment by mikepenz** [comment]
@StuStirling this seems to be invalid information inside the `*.pom` file of the dependency:

https://repo1.maven.org/maven2/org/glassfish/javax.annotation/10.0-b28/javax.annotation-10.0-b28.pom

```
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
<parent>
<groupId>org.glassfish</groupId>
<artifactId>javaee-api</artifactId>
<version>10.0-b28</version>
<relativePath>../pom.xml</relativePath>
</parent>
<modelVersion>4.0.0</modelVersion>
<groupId>org.glassfish</groupId>
<properties>
<extension.name>javax.annotation</extension.name>
<spec.version>1.0</spec.version>
</properties>
<artifactId>${extension.name}</artifactId>
<packaging>jar</packaging>
<name>${extension.name} API v.${spec.version}</name>
<description>Common Annotations for the JavaTM Platform API version ${spec.version} Repackaged as OSGi bundle in GlassFish</description>
<dependencies>
<dependency>
<groupId>javax.annotation</groupId>
<artifactId>jsr250-api</artifactId>
<version>${jsr250-api.version}</version>
<optional>true</optional>
</dependency>
</dependencies>
</project>
```

It's interesting as it seems to work with properties and placeholders. I'll investigate this particular one and see if we can build a common solution for those situations

**Comment by StuStirling** [comment]
Thanks. I'll look into overriding this specific one for now. FYI I was using an 8.0.0 release candidate prior to this which worked ok

**Comment by StuStirling** [comment]
@mikepenz sorry, struggling to exclude this library to then allow it to work. 

I've tried to follow the same format as in your sample `app` module's config file.

Here is the exclusion id I'm using: `org_glassfish__javax_annotation`.

I am then referencing this within a config directory at my project's root, like you're doing in the sample.

That right?

**Comment by mikepenz** [comment]
I think ultimately that one is super tricky because due to the invalid chars in the identifier various pieces wont' work out.

Locally I have the code working to apply those placeholders to artifactId, name and description. 

Should be able to provide you an update with this working in around 30-60 minutes

**Comment by mikepenz** [comment]
If you want to exclude it with the current version you may want to try to provide a `custom_exclusion_list.prop` via the config folder

with `org_glassfish__${extension_name}` as content

**Comment by mikepenz** [comment]
Please wait another 30 minutes until I was able to wrap up the release

**Comment by mikepenz** [comment]
When this build finishes things should be ready to go :) https://github.com/mikepenz/AboutLibraries/runs/1448519858?check_suite_focus=true

**Comment by StuStirling** [comment]
Great, thanks for such a quick turnaround. I can confirm that #565 has enabled the project to build again and provides the output.

```xml
<string name="define_plu_org_glassfish__javax_annotation" translatable="false"></string>
  <string name="library_org_glassfish__javax_annotation_libraryName" translatable="false">javax.annotation API v.1.0</string>
  <string name="library_org_glassfish__javax_annotation_libraryDescription" translatable="false"><![CDATA[Common Annotations for the JavaTM Platform API version 1.0 Repackaged as OSGi bundle in GlassFish]]></string>
  <string name="library_org_glassfish__javax_annotation_libraryVersion" translatable="false">10.0-b28</string>
  <string name="library_org_glassfish__javax_annotation_libraryArtifactId" translatable="false">org.glassfish:${extension.name}:10.0-b28</string>
```

May want to sort the placeholder in the `libraryArtifactId`? I guess it comes down to how far you want to go to fix someone else's bug!

Massive thanks for sorting this. Keep up the outstanding work

**Comment by mikepenz** [comment]
Will consider filling more placeholders tomorrow 😉

Thanks for the update. Glad it compile again

**Comment by mikepenz** [comment]
Released v8.6.2 with the remaining outlined problem. 

Thanks again for the report
